### ISSUE TITLE
Just a doubt with center_trim

### ISSUE THREAD

**Issue by Xirzag** [facebookresearch/demucs#122]
In the training code you use the function center_trim to translate the sources to compare them with the estimations returned by the network

```
streams = streams.to(device)
sources = streams[:, 1:]
sources = augment(sources)
mix = sources.sum(dim=1)

estimates = model(mix)
sources = center_trim(sources, estimates)
loss = criterion(estimates, sources)
```

Also, you use the center_trim inside the forward pass of the network.
So my question is, this function can produce a misalign from the output of the network and the target sources that can force the network to learn to translate the input function to match the target?
This misalignment can make problems with L1 or MSE loss functions that compares elements in the same position of the vector.

**Comment by adefossez** [comment]
Hey Xirzag,
I don't use padding at all in this code, but the length of the training size input is always chosen so that all the convolutions windows are full (so at each layer, the length of the input L is so that `(L - K) % S = 0` with `S`the stride of 4 and `K` the kernel size of 8). If it weren't for the `Conv1d`with a kernel size of 3 in the decoder layer, the output size would then be exactly the input size. Those `Conv1d` are not padded, so we loose one time step on each side. The `center_trim`in the forward is applied to the skip connection to remove those extra time steps present in the output of the encoder layer. Then I center trim the ground truth as well, given that the number of time steps lost on the left and right is the same.

Note that this is only valid at train time if the input length is a valid length as defined in the `valid_length`method in the `Demucs`class. At evaluation time, the input is always padded to reach a valid length if necessary.

**Comment by Xirzag** [comment]
Ok, now I understand it, 

thank you
